Add link resource

mwsteb opened the pull request.

Commit:
add migration and link model

Commit:
add migration and link model

Commit:
add link routes

Commit:
add link routes

Commit:
don't forget about controller!

Commit:
don't forget about controller!

Commit:
and templates with view!

Commit:
and templates with view!

mwsteb commented:
@azranel ready for review!

azranel commented:
Could you merge master and add `belongs_to :user, Shareguru.User` (or smth like that) to Shareguru.Link resource? 😄

Commit:
user has many links; link belongs to user

Commit:
user has many links; link belongs to user

azranel commented:
LGTM

Commit:
fix specs

Commit:
fix specs

mwsteb commented:
@azranel  should be cool now

Commit:
Change database config for dev and test env

Commit:
Merge pull request #6 from azranel/change_config

Change database config for dev and test env

Commit:
simplify Authentication Plug

let’s just assign user in conn, instead of using session - it’s easier
to mock signed in user that way

Commit:
fix more specs

we’re cheating AuthPlug by assigning current_user to conn

Commit:
Merge remote-tracking branch 'origin/add-link-resource' into add-link-resource

azranel commented on test/controllers/link_controller_test.exs:
```diff
@@ -0,0 +1,97 @@
+  test "lists all entries on index", %{conn: conn, user: user} do
+    conn = conn
+    |> assign(:current_user, user)
+    |> get(link_path(conn, :index))
```
This assigning could be extracted to ConnCase 😄

Commit:
BLARHGHA
